libs/features/mas/src/hydrate.js

```diff
@@ -275,43 +276,61 @@ export function updateLinksCSS(merchCard) {
     });
 }
 
+export function cleanup(merchCard) {
+  // remove all previous slotted content except the default slot
+  merchCard.querySelectorAll('[slot]').forEach((el) => {
+    el.remove();
+  });
+  const attributesToRemove = [
+  'checkbox-label',
+  'stock-offer-osis',
+  'secure-label',
+  'background-image',
+  'badge-background-color',
+  'badge-color',
+  'badge-text',
+  'size',
+  ANALYTICS_SECTION_ATTR,
+  ];
+  attributesToRemove.forEach(attr => merchCard.removeAttribute(attr));
+  const classesToRemove = ['wide-strip', 'thin-strip'];
+  merchCard.classList.remove(...classesToRemove);
+}
+
 export async function hydrate(fragment, merchCard) {
     const { fields } = fragment;
     const { variant } = fields;
     if (!variant) return;
-
-    // remove all previous slotted content except the default slot
-    merchCard.querySelectorAll('[slot]').forEach((el) => {
-        el.remove();
-    });
-
-    merchCard.removeAttribute('background-image');
-    merchCard.removeAttribute('badge-background-color');
-    merchCard.removeAttribute('badge-color');
-    merchCard.removeAttribute('badge-text');
-    merchCard.removeAttribute('size');
-    merchCard.classList.remove('wide-strip');
-    merchCard.classList.remove('thin-strip');
-    merchCard.removeAttribute(ANALYTICS_SECTION_ATTR);
-
+    merchCard.id = fragment.id;
+    // temporary hardcode for plans. this data will be coming from settings (MWPW-166756)
+    const settings = {
+      stockCheckboxLabel: 'Add a 30-day free trial of Adobe Stock.*', // to be {{stock-checkbox-label}}
+      stockOfferOsis: '',
+      secureLabel: 'Secure transaction' // to be {{secure-transaction}}
+    };
+    cleanup(merchCard);
```

Comment:
i would put clean up before id setup, you never know if some overzealous cleaner later will clean the id too

Reply:
true true